### ISSUE TITLE
HowTo

### ISSUE THREAD

**Issue by duczz** [MagnusPer/Balboa-GS510SZ#10]
Hey, i have this baboa Whirlpool too and wanna try these, but iam missing a HowTo.
Currently i have flashed your Project to a wemos Mini (with my settings) but i have no entitied in the homeassistant mqtt server.

What are ne next steps? Docker witch bwa?

**Comment by netmindz** [comment]
Have you tried the simple read example first to confirm the basic setup works first?

You don't mention if you have any existing Home Assistant setup already on place. You will need that and follow an appropriate guide to setup MQTT with Home Assistant

There are many different ways for this depending on how/where you are running HA

Once that's setup, you should see the spa under MQTT inside HA

**Comment by MagnusPer** [comment]
Hi and happy that you are interested in this library.

Firstly, Have you set your credentials in the code for the MQTT broker?
![IMG_0456](https://github.com/MagnusPer/Balboa-GS510SZ/assets/35903236/48d9ace2-b043-46b3-8aa0-0aa65c6f1aa8)

If so check that MQTT messages coming in from the SPA using a MQTT tool like http://mqtt-explorer.com/

**Comment by duczz** [comment]
Thanks for reply guys.
I have the Homeassitant Adoon MQTT Integration on my homeassitant, these i have enters already in the config, thats why i see via MQTT Explorer the Spa Parameters, but not on homeasssitant
![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/25475385/04074840-eb83-4ede-83b4-0ed458a4d712)

Edit: Settings are 0 because i have nothing on the esp currently
Edit2: it look like he is connected but i have no entities
![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/25475385/2b2d94b7-5d42-406a-bd6a-a554355e1cfa)

**Comment by MagnusPer** [comment]
Sorry, I do not have HA instead I use Homey so the integration to HA using MQTT I can not help you with

**Comment by duczz** [comment]
okay thank, so the next steps are only to let create the PCB layout and than put the j1 in the new medoul and j2 from the modul to j1 from the balboa right? not more to do?

edit: maybe the entities are not there because i have not connected to the balboa

**Comment by netmindz** [comment]
You could use with HA with just the raw MQTT, but it's *much* easier if you have a proper device registered in HA auto discovery, I've started and example

https://github.com/MagnusPer/Balboa-GS510SZ/pull/11

**Comment by netmindz** [comment]
Ok, that's the basic read-only example up there now

**Comment by duczz** [comment]
Perfect thanks, now i have the enteties.

![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/25475385/cf03a142-5519-4d39-a7f4-a1927e805515)

so now i need to let create the PCB layout and than put the j1 in the new modul and j2 from the modul back to j1 from the balboa right, than im finished?

**Comment by netmindz** [comment]
![Screenshot from 2023-09-02 18-13-48](https://github.com/MagnusPer/Balboa-GS510SZ/assets/442066/49b39c01-c70c-4078-b2df-f95117c2bddf)

**Comment by netmindz** [comment]
Latest version with buttons too - untested as I don't own that tub

**Comment by duczz** [comment]
to close the ticket, can someone answer my question?
so i need to let create the PCB layout and than put the j1 in the new modul and j2 from the modul back to j1 from the balboa right, than im finished?

can i control than the modies too and set the time for filter?

**Comment by MagnusPer** [comment]
@duczz - Between the display and the control system there is a cable with an RJ45 connector. Disconnect that from the Control system and place PCB in between. You need to create/buy the cable with RJ45 connectors between J2 -> Control system, NOTE that should have straight lines and not crossed as in a normal TP-cable.     

**DISPLAY**-------->(J1)  **PCB**  (J2)--------->**Control system**


There are no functions created to change modes or set filter times etc. The library however supports buttons to be pressed so theoretically it should be possible - see the section [button data ](https://github.com/MagnusPer/Balboa-GS510SZ#button-data)  

I hope this mini guide will help you to get it up and running

**Comment by duczz** [comment]
Perfect thank you Magnus.
@netmindz can you add all Buttons on this display to homeassitant? Than i can create automatisations to set mode / filtertimes etc.

![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/25475385/5f6f9ad4-c4ec-4b88-9791-c1be6b2cc3fc)

**Comment by netmindz** [comment]
I won't be adding any more to the code, if the buttons are supported by the library then it should be easy to follow the current examples.. If not, then update the library part of the code too

Create a fork then and then submit a PR

**Comment by duczz** [comment]
@netmindz alright, thanks for the first homeAsssitant port steps.
@MagnusPer are you able to update these? it looks like all buttons would be exist. i'm to noobie to port these :(
![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/25475385/6c8c3f93-da30-4ae7-9ca3-52fc481399c6)

**Comment by MagnusPer** [comment]
Sorry, I don’t have the knowledge how the Home Assistant works. The library support the “Mode” button and what I see is included in the HA example. The only missing piece is to add a HA button to trigger the mode.

**Comment by knogler** [comment]
@duczz have you found a solution?

**Comment by netmindz** [comment]
To add a button for pump3, just copy and paste what's listed for pump2

```
pump2.setName("Pump2");
pump2Button.setName("Pump2");
pump2Button.onCommand(onButtonPress);

```

but name pump3. Do same for any other buttons you want such as Mode

**Comment by netmindz** [comment]
As well as the definition of he button 

`HAButton pump2Button("Pump2");`

**Comment by knogler** [comment]
Thanks for the answer, I probably could have done that.

Unfortunately, setting the temperature is not displayed. somehow the variable “setTemperature” is missing from the homeassist file. And unfortunately I don't know how to create this in order to be able to regulate it.

**Comment by duczz** [comment]
@knogler nope, i was not able to migrate it complet to HomeAsssitant. I think this will never be done ^^ (i'm no developer :/)
But when all other works you could made a feature request then it will be updated here

**Comment by netmindz** [comment]
Try this version @knogler  https://github.com/MagnusPer/Balboa-GS510SZ/pull/13

**Comment by duczz** [comment]
@knogler Could you test it successfully?

**Comment by J0E-Cool** [comment]
Hi @all,
just a dumb question ;)  and sorry to reuse this chat...
I've already built the circuit on a stripboard 🙈, but now I'm thinking about, how I could test it with the still installed display VL100 and control board GS100... because, I guess, I've to adapt the programming.
Do you have some tips, lessons learned, how you did/do that? 
Thanks :)

**Comment by knogler** [comment]
Sorry for my late response. and thanks for coding. Unfortunately I haven't had time to test yet. I also have to see how I can get the software modified so that it runs with Balboa-MGS504DZ with panel VL700S (MAY800D). As soon as I have more information, I'll let you know.

**Comment by netmindz** [comment]
The model of top panel shouldn't really matter too much as we are emulating a panel. So what matters most is compatibility with the controller and the protocol it's speaking. Different models in the same range should use the same protocol so it shouldn't need code change.

What you might find is that your tub might only have 1 pump not two, so you mind find that pressuring pump2 might trigger the blower, just as if you had the wrong overlay applied to your top panel for your particular configuration

**Comment by MagnusPer** [comment]
> Hi @ALL, just a dumb question ;) and sorry to reuse this chat... I've already built the circuit on a stripboard 🙈, but now I'm thinking about, how I could test it with the still installed display VL100 and control board GS100... because, I guess, I've to adapt the programming. Do you have some tips, lessons learned, how you did/do that? Thanks :)

[@JOE-cool](https://github.com/J0E-Cool)

1) Start to just connect the circuit board to see everything works by manually press the panel buttons.
2) Easiest is to use example code [Balboa_GS_Read](https://github.com/MagnusPer/Balboa-GS510SZ/tree/main/examples/Balboa_GS_Read) to monitor via the serial monitor in [Arduino IDE ](https://docs.arduino.cc/software/ide-v2/tutorials/ide-v2-serial-monitor) if data is observed.
3) Use the example code  [Balboa_GS_Read](https://github.com/MagnusPer/Balboa-GS510SZ/tree/main/examples/Balboa_GS_Read) to activate button presses via the serial monitor input according to the [code](https://github.com/MagnusPer/Balboa-GS510SZ/blob/main/examples/Balboa_GS_Read/Balboa_GS_Read.ino). Example is `4` to activate light 
4) Now its time for remote access and automation. Depending of your knowledge and needs you have either the MQTT or HomeAssistant to work with

Good luck!

**Comment by J0E-Cool** [comment]
@MagnusPer thank you, helps me a lot :) I hope I can continue this weekend, if the weather allows. 
For the automation part, I've still to decide which one would fit best (don't have any in place), but step-by-step ;)

**Comment by ggtimtom** [comment]
Hello, i also try to make my GS523DZ with Panel VL802DZ smart. I hope the protocol will be similar.
I will start when the weather is fine.

**Comment by malu254** [comment]
> Hello, i also try to make my GS523DZ with Panel VL802DZ smart. I hope the protocol will be similar. I will start when the weather is fine.

Hello,
I think you need the Version from here https://github.com/Shuraxxx/-Balboa-GS523SZ-with-panel-VL801D-DeluxeSerie--MQTT 
The Hardware which you need will be the same

**Comment by duczz** [comment]
i try to build curretnly my lovelance Layout for the Summer. :)
Is it possbile to switch all Buttons with on/off function into a switch? what i need to edit for that?

**Comment by MagnusPer** [comment]
I using NodeRed as dashboard for my home automation. In NodeRed I create buttons triggering the MQTT commands and subscribes to MQTT status. Similar should be possible to do using Lovelance.   

![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/35903236/787523ef-236c-4a83-b33b-6df7fca9ab0c)

![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/35903236/1597e425-edaa-4a45-83e6-cbcae6436722)

**Comment by duczz** [comment]
Thanksmagnus, I had hoped that we could change habutton to haswitch in the *.ino File direct but i dont know If this works. Maybe @netmindz know it? :) (i dont know If haswitch exists in this program language)

**Comment by netmindz** [comment]
Yes there is @duczz  - see https://github.com/netmindz/balboa_GL_ML_spa_control/blob/7e193eb2708c2803dda82cae5e14056a8bfcb445/sensor/src/sensor.cpp#L120

The key thing is if you aren't just a button, then you need to know the current state, so it's not just swapping the type, but still pretty simple to do. See standalone example https://github.com/dawidchyrzynski/arduino-home-assistant/blob/main/examples/light/light.ino

**Comment by netmindz** [comment]
I haven't tested the code, but it will be something along the lines of this https://github.com/netmindz/Balboa-GS510SZ/commit/4d32135e64c111bf3d1619eb5a832677b0603a93

**Comment by duczz** [comment]
i have created on netmindz top tipps and example (much thanks without i cant do this) a new version, @MagnusPer can you create a new "Balboa_GS_HomeAssistant_DEV.ino" with this content? Would be create when someone can test it, my platine is not delivered :( @J0E-Cool J0E-Cool 
https://pastebin.com/7bMn9JAM (maybe netmindz can make a quick look, how it looks)

Someone knows why line 237 is commented out? 

![image](https://github.com/MagnusPer/Balboa-GS510SZ/assets/25475385/e1f4ab11-68c2-42e6-b44f-faad934390e8)

**Comment by netmindz** [comment]
@duczz  - I'm not familiar with this specific controller, but I know on some spas the buttons have multiple functions, so even if there are switches for the pumps, it might not be wise also to remove the ability to send button presses, unless you know for sure that you only need mode and temp +/-

As for your onSwitchCommand, this needs to look to see **which** switch you are trying to change and then read the right state, you currently are reading the state of the light.

For example of checking the sender see https://github.com/netmindz/balboa_GL_ML_spa_control/blob/7e193eb2708c2803dda82cae5e14056a8bfcb445/sensor/src/sensor.cpp#L240

**Comment by MagnusPer** [comment]
@duczz - When you have a working and tested example you can do a Pull Request and then I can approve it to be added to this repository

**Comment by duczz** [comment]
okay thanks netminz, i tried, but that programming is to high for me im out ^^

**Comment by awallace24335** [comment]
> @duczz - Between the display and the control system there is a cable with an RJ45 connector. Disconnect that from the Control system and place PCB in between. You need to create/buy the cable with RJ45 connectors between J2 -> Control system, NOTE that should have straight lines and not crossed as in a normal TP-cable.
> 
> **DISPLAY**-------->(J1) **PCB** (J2)--------->**Control system**
> 
> There are no functions created to change modes or set filter times etc. The library however supports buttons to be pressed so theoretically it should be possible - see the section [button data ](https://github.com/MagnusPer/Balboa-GS510SZ#button-data)
> 
> I hope this mini guide will help you to get it up and running

Just to confirm, can this be a normal patch cable or do the pairs have to be untwisted? 

Thank you

**Comment by knogler** [comment]
yes, it's a normal patch cable

**Comment by jarnodrenth** [comment]
> Sorry, I do not have HA instead I use Homey so the integration to HA using MQTT I can not help you with

 Can you tell me what you use for homey?
